File: scratch/fix-fe.py

import re

# ManajemenLokasi.tsx
with open('frontend/src/pages/ManajemenLokasi/ManajemenLokasi.tsx', 'r', encoding='utf-8') as f:
    c = f.read()
c = re.sub(r'const _rwGroups = useMemo\(\(\) => \{.*?\}, \[householdGroups, locations\]\);', '', c, flags=re.DOTALL)
c = re.sub(r'formatRegionName\(loc\.rw, loc\.kelurahan\)', 'formatRegionName(loc.rw)', c)
with open('frontend/src/pages/ManajemenLokasi/ManajemenLokasi.tsx', 'w', encoding='utf-8') as f:
    f.write(c)

# RwApproval.tsx
with open('frontend/src/pages/RwPortal/RwApproval.tsx', 'r', encoding='utf-8') as f:
    c = f.read()
c = re.sub(r'const \[_pendingBins, _setPendingBins\] = useState<any\[\]>\(\[\]\);\n', '', c)
c = re.sub(r'setPendingBins\(binsRes\.data\);', '//', c)
c = re.sub(r'const _approveBin = async.*?\};\n', '', c, flags=re.DOTALL)
c = re.sub(r'const _rejectBin = async.*?\};\n', '', c, flags=re.DOTALL)
with open('frontend/src/pages/RwPortal/RwApproval.tsx', 'w', encoding='utf-8') as f:
    f.write(c)

# MasterQrManager.tsx
with open('frontend/src/pages/SuperAdmin/MasterQrManager.tsx', 'r', encoding='utf-8') as f:
    c = f.read()
c = re.sub(r'const \[_pendingBins, _setPendingBins\] = useState<any\[\]>\(\[\]\);\n', '', c)
c = re.sub(r'setPendingBins\(pendingBinsRes\.data\.data\);', '//', c)
c = re.sub(r'const _approveBin = async.*?\};\n', '', c, flags=re.DOTALL)
with open('frontend/src/pages/SuperAdmin/MasterQrManager.tsx', 'w', encoding='utf-8') as f:
    f.write(c)

File: scratch/fix-fe.ts

const fs = require('fs');

let f1 = fs.readFileSync('frontend/src/pages/ManajemenLokasi/ManajemenLokasi.tsx', 'utf8');
f1 = f1.replace(/const _rwGroups = useMemo\(\(\) => \{[\s\S]*?\}, \[householdGroups, locations\]\);/g, '');
f1 = f1.replace(/formatRegionName\(loc.rw, loc.kelurahan\)/g, 'formatRegionName(loc.rw)');
fs.writeFileSync('frontend/src/pages/ManajemenLokasi/ManajemenLokasi.tsx', f1);

let f2 = fs.readFileSync('frontend/src/pages/RwPortal/RwApproval.tsx', 'utf8');
f2 = f2.replace(/setPendingBins\(binsRes.data\);/g, '// setPendingBins(binsRes.data);');
f2 = f2.replace(/const _approveBin = async[\s\S]*?  \};\n/g, '');
f2 = f2.replace(/const _rejectBin = async[\s\S]*?  \};\n/g, '');
fs.writeFileSync('frontend/src/pages/RwPortal/RwApproval.tsx', f2);

let f3 = fs.readFileSync('frontend/src/pages/SuperAdmin/MasterQrManager.tsx', 'utf8');
f3 = f3.replace(/setPendingBins\(pendingBinsRes.data.data\);/g, '// setPendingBins(pendingBinsRes.data.data);');
f3 = f3.replace(/const _approveBin = async[\s\S]*?  \};\n/g, '');
fs.writeFileSync('frontend/src/pages/SuperAdmin/MasterQrManager.tsx', f3);

